File: test/leetcode/graph/nested-list-weighted-sum.test.ts
import { NestedInteger } from '../../src/linked-list/common/nested-integer';
import { depthSum } from '../../src/graph/nested-list-weighted-sum';

describe('nested list weighted sum', () => {
  test('nested list weighted sum - test case 1', async () => {
    expect(depthSum([new NestedInteger(10)])).toBe(10);
  });

  test('nested list weighted sum - test case 2', async () => {
    expect(depthSum([new NestedInteger(10), new NestedInteger(20)])).toBe(30);
  });
});

File: test/leetcode/graph/nested_list_weighted_sum_test.py
from leetcode.graph.common.nested_integer import NestedInteger
from leetcode.graph.nested_list_weighted_sum import Solution


soln = Solution()


def test_case_1():
    assert soln.depthSum([NestedInteger(10)]) == 10


def test_case_2():
    assert soln.depthSum([NestedInteger(10), NestedInteger(20)]) == 30
